### PR TITLE
chore: use smaller model for planner

### PR DESCRIPTION
**Pull request by muralov**
**Description**
Use smaller models (like gpt-4o-mini) for planner

Changes proposed in this pull request:

- Use smaller model for planner node
- Implement integration tests with DeepEval
- Refactor loading configs
- Refactor loading env vars

**Related issue(s)**
https://github.com/orgs/kyma-project/projects/61/views/1?pane=issue&itemId=80761458

### CODE REVIEW

File: src/agents/graph.py [hunk 1]
```diff
@@ -81,21 +81,30 @@ class KymaGraph:
...
+    def __init__(self, models: dict[str, IModel], memory: BaseCheckpointSaver):
+        self.models = models
         self.memory = memory
 
-        self.kyma_agent = KymaAgent(model)
-        self.k8s_agent = KubernetesAgent(model)
+        gpt_4o_mini = models[LLM.GPT4O_MINI]
+        gpt_4o = models[LLM.GPT4O]
+
+        # TODO: switch to gpt_4o when necessary
```
Comment (friedrichwilken) [inline src/agents/graph.py:93]: Is this really a todo, a task we will resolve in the future?

File: tests/integration/agents/test_graph.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        evaluation_params=[
+            LLMTestCaseParams.ACTUAL_OUTPUT,
+            LLMTestCaseParams.EXPECTED_OUTPUT,
+        ],
+        model=evaluator_model,
+        threshold=0.85,
+    )
+
+
+def create_state(messages: Sequence[BaseMessage]) -> AgentState:
```
Comment (friedrichwilken) [inline tests/integration/agents/test_graph.py:38]: Can please add a comment that explains how this works?

File: tests/integration/agents/conftest.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+from threading import Thread
+
+import pytest
+from deepeval.models.base_model import DeepEvalBaseLLM
+from fakeredis import TcpFakeServer
+
+from agents.graph import KymaGraph
+from agents.memory.redis_checkpointer import RedisSaver, initialize_async_pool
+from utils.models import LLM, ModelFactory  # noqa: E402
```
Comment (friedrichwilken) [inline tests/integration/agents/conftest.py:9]: Why do we have a `# noqa: E402` here?

File: tests/integration/agents/test_graph.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                    content="The user query is related to: "
+                    "{'resource_api_version': 'v1', 'resource_namespace': 'nginx-oom'}"
+                ),
+                HumanMessage(content='Write "Hello, World!" code in Python'),
+            ],
+            'Here is a simple "Hello, World!" program in Python: `print("Hello, World!")`',
+        ),
+    ],
+)
+def test_common_node(messages, expected_answer, kyma_graph, answer_relevancy_metric):
```
Comment (friedrichwilken) [inline tests/integration/agents/test_graph.py:229]: can you please add a comment that explains what the purpose of the test is, what is done etc.?

File: tests/integration/agents/test_graph.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+                    content="Create a hello world app with python and deploy it with Kyma?"
+                ),
+            ],
+            '{"subtasks": [{"description": "Create a hello world app with python", "assigned_to": "Common"},'
+            '{"description": "deploy it with Kyma", "assigned_to": "KymaAgent"}]}',
+            False,
+        ),
+    ],
+)
+def test_planner(
```
Comment (friedrichwilken) [inline tests/integration/agents/test_graph.py:174]: can you please add a comment that explains what the purpose of the test is, what is done etc.?

**Review by friedrichwilken — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-test.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      - "config/**"
+      - "data/**"
+      - "scripts/**"
+
+## **IMPORTANT**: If any changes are made to how to run the integration tests. Make sure to update the steps for integration-tests
+## in the create-release.yml workflow as well.
+
+jobs:
+  int-test:
+    if: contains(github.event.pull_request.labels.*.name, 'run-int-tests')
```
[the remark this answers is not in the document]
Comment (muralov) [reply .github/workflows/integration-test.yaml:24]: it seems it is convention to have "-" sign if contains multiple words. 
I can name it "run-integration-tests". Is it okay for you?

File: src/agents/graph.py (hunk 1 above)
Comment (muralov) [reply src/agents/graph.py:93]: we need to change it to gpt_4o as we implement Kyma agent. I think, gpt_4o is not enough for such complex task

Comment (friedrichwilken) [reply src/agents/graph.py:93]: Okay, that makes sense.

File: .github/workflows/integration-test.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (friedrichwilken) [reply .github/workflows/integration-test.yaml]: I would prefer it `"run-integration-tests"` because of the ambiguity of int.

**Review by friedrichwilken — APPROVED** [review]

**Review by friedrichwilken — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**